### ISSUE TITLE
Paper: MOAB Fusion Network Experiment

### ISSUE THREAD

**Issue by Sukh-P** [openclimatefix/pvnet#272]
## Arxiv/Blog/Paper Link

Paper from Queen Mary's University (@omniaalwazzan) can be found [here](https://arxiv.org/pdf/2403.06349v1) and code related to the paper [here](https://github.com/omniaalwazzan/MOAB)

## Detailed Description

This paper from Queen Mary's University DERI group presents a new type of fusion architecture (Multi-Modal Outer Arithmetic Block, MOAB) which is fusion network which uses outer arithmetic (e.g. outer product, addition etc) to create interaction vectors, this is applied to a problem which has similarities to PVNet e.g. multimodal inputs and separate encodings before fusion of the 1d vectors. 

Would be an interesting experiment to implement and train a MOAB style fusion network instead of the concatenation fusion network we currently use to compare forecasting results (would be good to compare other fusion networks too).

**Comment by Sukh-P** [comment]
Hey @omniaalwazzan, thanks so much for your message and for taking a look! This was a quick and dirty implementation but I wanted to get some results to see how it was doing, I used the MOAB technique on our two main inputs which are a numerical weather prediction and satellite images as they are the two most different types of modalities we pass in, they then combine with some other more minor inputs id embeddings and some sun position encodings. 

Yes the learning rate drop didn't kick in as I had not set the early stopping patience > the patience on the LR scheduler, need to change that for the next run. 

Yes the weight decay is heavy, we have had good results from using this but I could play with that hyperparameter. And yes I didn't overwrite some placeholder directories but actually I think that warning is unrelated as even when I have changed it it appears, one to look into but not impacting things currently as far as I can see. 

Thanks for your kind offer, when I get a bit of time to revisit and rerun this I will update this issue and let if you know if I have any issues/blockers!
